src/pages/journal/__tests__/journal.effects.spec.ts

```diff
@@ -135,13 +135,16 @@ describe('Journal Effects', () => {
       ConnectionStatus.ONLINE,
       false,
       new Date())); // Load in mock journal state
-    spyOn(store$, 'dispatch');
     // ACT
     actions$.next(new journalActions.SelectNextDay());
     // ASSERT
     effects.selectNextDayEffect$.subscribe((result) => {
-      expect(store$.dispatch).toHaveBeenCalledWith(new journalActions.JournalNavigateDay(nextDay));
-      expect(result).toEqual(new journalActions.SetSelectedDate(nextDay));
+      if (result instanceof journalActions.SetSelectedDate)  {
```

Comment:
do we need these ifs?

Reply:
I feel that this is a hackie solution. But couldn't do it better. So the function in subscribe will be called twice because of the elements of the action array will be arrive one by one. First time the first if will be true so we can make the right assertion and second time the second.

Let's have a think about this in the future and come up with a better solution :)